softwareComponents/configuration/include/configuration/serialization.hpp

```diff
@@ -250,26 +277,30 @@ namespace rofi::configuration::serialization {
             j[ "point" ][ "z" ] = sj.refPoint[ 2 ];
             assert( sj.joint.get() && "joint is nullptr" );
             j[ "joint" ] = jointToJSON( *sj.joint );
+            addAttributes( j, attrCb, sj );
             res[ "spaceJoints" ].push_back( j );
         }
 
         return res;
     }
 
-    template<>
+    inline nlohmann::json toJSON( const Rofibot& bot ) {
+        return toJSON( bot, [](auto ...){ return nlohmann::json{}; } );
+    }
+
     inline Rofibot fromJSON( const nlohmann::json& j ) {
```

Comment:
`fromJson` should take a lambda as well - so that the user doesn't have to parse the json himself to get the attributes.